### PR TITLE
Add a DELETE api for CloudAuthz

### PR DESCRIPTION
**Pull request by VJalili**

### CODE REVIEW

File: lib/galaxy/webapps/galaxy/api/cloudauthz.py [hunk 1]
```diff
@@ -146,3 +146,29 @@ def create(self, trans, payload, **kwargs):
...
+        """
+        * DELETE /api/cloud/authz/{encoded_authz_id}
+            Deletes the CloudAuthz record with the given ``encoded_authz_id``.
+        """
+
+        msg_template = "Rejected user `" + str(trans.user.id) + "`'s request to delete cloudauthz config because of {}."
+        try:
+            authz_id = self.decode_id(encoded_authz_id)
+        except Exception:
+            log.debug(msg_template.format("cannot decode authn_id `" + str(authz_id) + "`"))
```
Comment (dannon) [inline lib/galaxy/webapps/galaxy/api/cloudauthz.py]: You probably mean for this to be `encoded_authz_id`?

Comment (dannon) [reply lib/galaxy/webapps/galaxy/api/cloudauthz.py]: (and in the line below, `authz_id`, instead of `authn_id`?)

Comment (VJalili) [reply lib/galaxy/webapps/galaxy/api/cloudauthz.py]: good catch, I guess too many `authz`, `authn`, and `authnz` ;-)

**Comment by VJalili** [comment]
ping @Nerdinacan I am adding an api to delete a cloudauthz record with this PR, and I will make another one PR for update API.

**Review by jmchilton — APPROVED** [review]

**Comment by VJalili** [comment]
Thanks @dannon

**Pull request merged**